### PR TITLE
Add basic sentry/prometheus metrics

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: sygnal/__init__.py [hunk 1]
```diff
@@ -287,6 +314,17 @@ def setup():
...
+    if cfg.has_option("metrics", "sentry_dsn"):
+        # Only import sentry if enabled
+        import sentry_sdk
+        sentry_sdk.init(
+            dsn=cfg.get("metrics", "sentry_dsn"),
+            integrations=[sentry_sdk.integrations.flask.FlaskIntegration()],
+        )
+
+    if cfg.has_option("metrics", "prometheus_port"):
+        prometheus_client.start_http_server(cfg.get_int("metrics", "prometheus_port"))
```
Comment (michaelkaye) [inline sygnal/__init__.py]: SHould include address to listen on pls :/

File: sygnal/__init__.py [hunk 2]
```diff
@@ -26,16 +26,36 @@
...
+
 import sygnal.db
 from sygnal.exceptions import InvalidNotificationException
 
+
+NOTIFS_RECEIVED_COUNTER = Counter(
+    "sygnal_notifications_received", "Number of notification pokes received",
+)
+
+NOTIFS_RECEIVED_COUNTER = Counter(
```
Comment (richvdh) [inline sygnal/__init__.py]: one of these needs a different name

File: sygnal/__init__.py [hunk 3]
```diff
@@ -26,16 +26,36 @@
...
+    "sygnal_notifications_devices_received",
+    "Number of devices been asked to push",
+)
+
+NOTIFS_BY_PUSHKIN = Counter(
+    "sygnal_per_pushkin_type",
+    "Number of pushes sent via type of pushkin",
```
Comment (richvdh) [inline sygnal/__init__.py]:
```suggestion
    "Number of pushes sent via each type of pushkin",
```

File: sygnal/__init__.py [hunk 4]
```diff
@@ -287,6 +314,17 @@ def setup():
     else:
         logging.basicConfig()
 
+    if cfg.has_option("metrics", "sentry_dsn"):
```
Comment (richvdh) [inline sygnal/__init__.py]: is there not some "default config" or "sample config" that this needs adding to?

Comment (erikjohnston) [reply sygnal/__init__.py]: I've added it to sample config.

**Pull request merged**